tests/test_requirements.py

```diff
@@ -1195,3 +1200,21 @@ def test_try_star_requirements():
         "otherwise",
         "finish",
     ]
+
+
+@type_parameter_syntax
+def test_type_var_requirements():
+    node = _parse("type Alias[T: (str, bytes)] = list[T]")
+    assert _dep_names(node) == ["str", "bytes", "list", "T"]
```

Comment:
My reading is that `type` introduces a new scope.  I think you're going to have to do something like what we do for functions to prevent `T` from appearing in this list.

In the following example, the first statement should not depend on the definition of `T` in the second statement as the binding should be shadowed by the type variable:
```python
type Alias[T] = list[T]
T = 4
```

Reply:
Hi @bwhmather. Good point! Let me fix this.

Reply:
This is going to be tricky because we don't have enough context when parsing the right side of the expression. It can be `list[T]` where T is a placeholder for a generic type or a `list[MyClass]` where MyClass is an actual type with many kinds of more complex variations... Any idea how CPython differentiates one from the other while parsing?